internal/metrics/remote_write.go

```diff
@@ -0,0 +1,394 @@
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"math"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+
+	"github.com/golang/snappy"
+	"github.com/project-flotta/flotta-operator/models"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+)
+
+const (
+	defaultWaitInterval         = 5 * time.Minute
+	defaultRequestNumSamples    = 10000
+	defaultRequestDuration      = time.Hour
+	defaultRequestRetryInterval = 10 * time.Second
+	LastWriteFileName           = "metrics-lastwrite"
+	DeviceLabel                 = "edgedeviceid"
+)
+
+type RemoteWrite struct {
+	deviceID             string
+	config               RemoteWriteConfiguration
+	tsdb                 API
+	LastWrite            time.Time
+	lastWriteFile        string
+	waitInterval         time.Duration
+	RangeDuration        time.Duration
+	RequestNumSamples    int
+	RequestRetryInterval time.Duration
+	Client               WriteClient
+}
+
+type RemoteWriteConfiguration struct {
+	Url     string // empty URL means that feature is disabled
```

Comment:
Let's make sure it is documented.

Reply:
sure. I'll add a story for documentation.

Reply:
https://issues.redhat.com/browse/ECOPROJECT-677